timesketch/lib/datastores/opensearch.py

```diff
@@ -1237,3 +1247,290 @@ def flush_queued_events(self, retry_count=0):
+                es_logger.error(f"[PIT Slice {slice_id+1}/{num_slices}] Failed to put sentinel: Queue full.")
+            except Exception as e_s:
+                es_logger.error(f"[PIT Slice {slice_id+1}/{num_slices}] Failed to put sentinel: {str(e_s)}")
+
+
+    def export_events_pit(
+        self,
+        indices_for_pit: List[str],
+        base_query_body: Dict[str, Any],
+        sort_criteria: Optional[List[Dict[str, Any]]] = None,
+        page_size: int = 10000,
+        pit_keep_alive: str = "5m",
+        num_slices: int = 4,
+        request_timeout_per_slice: Optional[int] = None
+    ) -> Generator[Dict[str, Any], None, None]:
+        """
+        Exports events from specified indices using PIT, search_after, and slicing.
+
+        This method is designed for efficient bulk export of large datasets.
+        It yields individual event dictionaries, including '_id' and '_index'.
+
+        Args:
+            indices_for_pit: List of index names to query.
+            base_query_body: The base OpenSearch query body (must not include
+                             size, sort, pit, or slice clauses). Can include
+                             "_source" to specify fields to return.
+            sort_criteria: The sort criteria for search_after.
+                           If None, defaults to `[{"_id": "asc"}]`.
+                           Must include a unique tie-breaker field (e.g. _id)
+                           for stable results across pages.
+            page_size: Number of documents per request per slice. Default: 10k
+            pit_keep_alive: Keep-alive duration for the Point-In-Time context.
+            num_slices: Number of parallel slices to use for the export.
+                        Default can be set via app config OPENSEARCH_PIT_NUM_SLICES.
+            request_timeout_per_slice: Optional timeout in seconds for individual
+                                       OpenSearch search requests within each slice.
+                                       Defaults to self.timeout + 20s buffer.
+
+        Yields:
+            Dict: Individual event documents (including _id and _index from the hit).
+
+        Raises:
+            ValueError: If indices_for_pit is empty or num_slices is invalid.
+            errors.DataStoreQueryError: For critical issues during the export.
+        """
+        if not indices_for_pit:
+            raise ValueError("indices_for_pit cannot be empty.")
+        if not 1 <= num_slices <= 1024:  # OpenSearch constraint for slices
+            raise ValueError("num_slices must be between 1 and 1024.")
+
+        effective_sort_criteria = sort_criteria or _DEFAULT_PIT_SORT_CRITERIA
+        worker_timeout = request_timeout_per_slice or (self.timeout + 20) # Use configured or default+buffer
+
+        # Adjust num_slices from app config if not provided or if a default is desired.
+        # Example: num_slices = num_slices or current_app.config.get('OPENSEARCH_PIT_NUM_SLICES', 5)
```

Comment:
shall we remove that if we do not use a Config variable here?

Reply:
This value should be configurable for sure. The question is where we consume the config. 
I would probably opt for a solution that uses the value from the configuration as default but still can be overwritten by a caller if necessary. What do you think?

Reply:
how about we give it a Default here and if a config value is present to use the config?

Reply:
I went with the following priority: Caller arg > timesketch.conf setting > default value